### DISCUSSION TITLE
End of Maintenance Notice for Skeleton v2

### DISCUSSION THREAD

**Discussion by endigo9740** [skeletonlabs/skeleton#3063] [Announcements]
> NOTE: Updated March 4, 2025 to better express the expected launch window

Hey everyone, Chris here from Skeleton. As we move closer to the upcoming release of Skeleton v3, we wanted to layout explicit plans for Skeleton v2 going forward. Let's begin by reviewing what is changing and why.

## Release Plans

We've created a diagram to help summarize the changes.

![Screenshot 2024-12-18 at 12 16 00 PM](https://github.com/user-attachments/assets/35cd0767-18e7-47af-8bb7-5f3c45f64754)

Essentially this means Skeleton v3 will be moved from an unstable pre-release to our active production branch during the v3 release. At this same time, Skeleton v2 will move from it's current maintenance mode, to being fully archived. For us, archived means there will be no more code changes and no more package releases from that point forward. As far as we're concerned it will be fully sunset.

## Can I still use Skeleton v2?

Absolutely. Your current installation of v2 will continue to operate as it does now, but will no longer receive updates from that point forward. All NPM packages will remain available - though you may have to specify the version on install. And we'll preserve a backup of the source code on GitHub via a dedicated branch, [just like Skeleton v1](https://github.com/skeletonlabs/skeleton/tree/v1).

Additionally we will continue to host the v2 documentation, which will follow our standard subdomain pattern:

| Domain                     | Role                     |
| -------------------------- | ------------------------ |
| https://v1.skeleton.dev/   | v1 documentation archive |
| https://v2.skeleton.dev/   | v2 documentation archive |
| https://skeleton.dev/      | v3 documentation         |
| https://next.skeleton.dev/ | Reserved for Skeleton v4 |

And again, just like our previous transitions, we'll provide a quick link to the older archived versions on the new site:

![Screenshot 2025-03-04 at 4 07 22 PM](https://github.com/user-attachments/assets/69faa58e-e65d-4899-bece-aec425770239)

## Why is v2 being archived?

As you may know, we're a small team, which means we don't have the resources to maintain multiples versions of Skeleton in parallel. We actually tried our best to maintain v2 and v3 at the same time this past year, but did eventually [shift priorities](https://github.com/skeletonlabs/skeleton/discussions/2729) to focus our full attention on Skeleton v3. Since then, all contributions to v2 have come directly from the community.

## When will this change occur?

As of [this announcement](https://github.com/skeletonlabs/skeleton/discussions/3315), Skeleton v2 is no longer receiving new updates. We'll begin moving to archive it over the next few days.

## What happens to v2 issues on GitHub?

The [Skeleton v2 milestone](https://github.com/skeletonlabs/skeleton/milestone/4) is now closed and we're now longer accepting new issues.

## What about the Create-Skeleton-App (CLI)

Alongside the v3 launch, we will continue to provide the CSA, which is used for scaffolding new Skeleton v2 projects. However, we will be taking steps to move this to maintenance mode asap. As part of this process we will be reviewing options to keep this permanently available, though it will be provided "as is" with no further maintenance going forward.

However, we've recently introduced it's successor, the [Skeleton CLI](https://github.com/skeletonlabs/skeleton/pull/3091). This will initially be limited to a single purpose - aiding in migrating to Skeleton v3. However, we have plans to expand this with new features over time.

## Will any v2 issues move to v3?

In _most_ cases it will not be relevant for v2 issues to carry over to v3. This is due to the fact that most components and features have been rebuilt from the ground up. While they may have new issues to be found and reported, the previous issues are likely no longer relevant. However, in the odd case that something is relevant (a feature request for example) we will consider moving this forward.

## Can I migrate from Skeleton v2 to v3?

Yes, please [follow this migration guide](https://next.skeleton.dev/docs/get-started/migrate-from-v2).

## Questions and Feedback

If you have any questions about these changes or would like to provide feedback, please reach out in the comment section below. The team and I are also available via the `#skeleton-next` channel on Discord.

**Comment by stalkerg** [comment]
What about critical issues? Do you plan to provide deps udpate at least this year? 
I mean, we can't migrate overnight (we still convert maybe only 30% of project to Svelte5), and mantain deps overwrite it's annoying. I understand what you are small team, but maybe with extra help it is posible? 
btw what about EOL for tailwind?

**Reply by endigo9740** [reply]
In the case of an absolute emergency like a security risk or something that would render Skeleton v2 completely unusable we would make changes to address that, but it would be an exception, not the norm. Regular maintenance from the Skeleton team itself ended for v2 months ago. The small handful of updates we've seen the past few months have been from the community, mostly focusing on bug fixes and text adjustments in the documentation. It's not perfect, but v2 is relatively stable for what it is.

Not sure what you mean by Tailwind - if you mean which version supports which. v2 will be the last release of Skeleton that officially supports Tailwind v3, and Skeleton v3 will be the first version that supports Tailwind v4. I expect v4 will stick around a while, with major releases being much smaller for a bit (not another total rewrite).

The thing to keep in mind is the last year has been extremely volatile for major updates to the underlying technology we reply on. Especially Svelte 5 and Tailwind v4. So a lot of these changes are just part of the process of keeping things up to date. I do foresee an extended period of time where things settle out and we see much more incremental updates for an extended period (years even). But we've gotta get v3 out the door to begin that process.

**Reply by endigo9740** [reply]
Also...

> but maybe with extra help it is possible

Everyone that contributes is a volunteer on this project, the team and myself included. Anyone that wishes to see more changes should try to find the time to contribute. That's how open source works after all.

- I devote half of my working hours every week (20+ hours a week) to being seat in the chair working on Skeleton. And basically every other moment of free time providing support (such as responding to this message)
- @Hugos68 devotes nearly that much while juggling work and school
- @phamduylong submits PRs and bug fixes nearly daily
- The various other team members provide as much support as they can on Discord, etc
- And we're lucky to have a handful of regular contributors from the community.

Even with all this, this is not enough to maintain two major release of our library. I know bigger companies with multiple full time staff members that couldn't pull something like that off. I'm sure your comment is well intended, but unfortunately it's just not in the cards for us.
